### PR TITLE
Work up to November 12

### PR DESCRIPTION
**Pull request by DanielTurner**
UAT (https://www.qld.gov.au/dev/food-pantry/uat/label-buster) for Label Buster has been updated. 
* Ingredient page content.
* Addition of ingredient list table component.
* Help guide on ingredient page.
* Help guide on Business details page.
* Help guide updates from last UAT feedback.
* PDF and email generation on Your label page.
* Content update to Your Label to support dynamic summary (the data for this will be progressively added in future sprints).
* Food Pantry Form Wrapper and config updates to enable code re-use across the two products.
Still in progress are:
* Formatting of PDF content.
* Business and details form content

Code changes are as follows:
* Updated backups of form.json
* Updated karma config for testing.
* Added post install hook to fix issue with parcel build
* Added button group component.
* Added formio wrapper component
* Added help guide component
* Added partials for help guide
* Added config for wrapper
* Added environment for code
* Added index to for entrypoint for SPA
* Added js to work with that index.html
* Added resources for formio html parts
* Added in expand/collapse script to fill in for functionality from SWE not working
* Added in styles
* Added in tests for button group
* Added in tests for formio wrapper events
* Added in test for formio wrapper

### CODE REVIEW

File: src/components/partials/help-guide-lb-business.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import { html } from 'lit-html';
```
Comment (duttonw) [inline src/components/partials/help-guide-lb-business.js:1]: what does this file do? should the data not be inside form.io and a shim is used to make it pretty?

File: src/resources/pdf/styles.scss [hunk 2]
```diff
@@ -0,0 +1,220 @@
+@media print {
```
Comment (duttonw) [inline src/resources/pdf/styles.scss:1]: @asifamingov  can you review this.

File: src/resources/questions-ingredients/question-1/view.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<small class="hint"
```
Comment (duttonw) [inline src/resources/questions-ingredients/question-1/view.html:1]: @DanielTurner  why is this not inside the form.io page?

File: src/resources/your-label/section-one/styles.scss [hunk 4]
```diff
@@ -0,0 +1,66 @@
+.lb {
```
Comment (duttonw) [inline src/resources/your-label/section-one/styles.scss:1]: @ganapavz  @asifamingov 
is the styles here suitable, should we look at including some of these into the swe?

File: src/styles/formio/radio.scss [hunk 5]
```diff
@@ -0,0 +1,66 @@
+.formio-component-radio {
```
Comment (duttonw) [inline src/styles/formio/radio.scss:1]: @asifamingov  should we look at including this into the swe?

File: json/PDF/form.json [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
.../sakKxgHOgXSOefCW+zRt0w3Jqjb9yu0htsdvH1P3LXYbyXjhWJiFbA/gznT3GtU/uBpocFwOCw+5RUmH+gnjisc9aV9V4K17MdxLv9JFBmhgZVsStLJsYgEBR3FoU7QgSZaLYUQUsXyst2dNoGbCNZ7SoSjYHAMX3jN9KildWHeqGUdcLwkignHf35rUv5aMIgqq/OsVV2CTIaINAQdxLGEcTJSJWTasn8mAvySurAKAKYdbvz8JAkYN0jKvI7t6pm6Wc4Yt9nLFdLPgfFbk65hpCMYE+DjBb5FQWcRi4lO1ax4qnaHLlmvPEqRE5FQB5FALyqDKq6qtmUPvvmhHABY31QK61a2PLZfbAg1aFd6YszPgfFbk86n14WviHGd0lIIfI7dygY6i8Tpfsq2PyasiwTsyMsRoKKkEhy+rJqs3izJ8y/N2GPadnlCtb25xUiGnwTjNyadT0+zl7m+IzL6TLVwULAlFToLQa6XbawSUSwlAo2ImODgPJH1IfgX5pyzm1wMWu8nOnVubIX6lKKcKsbVtLHxj8P4jUmX08deG96UQu1a4PjHxPrbA3kD9O+7C10Zh/cUNNDe9OwYaHd2uznAi8LYoQWYjvvt2+HkBz+GDie/Vtr1HbT6yT7kyybtD5+Ht8psSEE+SY14Hqnh8JXfGWuIvJSWV/ZZCK9h3/mv1H43zq0yJ6UCvkrFrHytEifqkpbLjgA2FF8djEe13+1uUqwxRV/0pxu+/rO/QzcljUvoMWeQe/fTI95/dLr/JCT8rphkIX74cmj8imDU0M0iYnzo7UO1H8jS0PGh14MBgFYP+8KBDA+42xM/QNnuQNUD60NfHbpZAu86fh0w/nksVkuEur8FPRFoYLrB/h786b6AcnPA3TMDAN0zeqLnu+cH/awxDKABjwtD8oTXCSOuFmLlSU2cgR3wjJ64KGlrHtWsvZ9G5mVYCK8P9zN78VqE8QTGhXTc09t7NTz8G5/b2+s9Xd0eBjwAGB7f4tANQ2ECHf7dDf3hkTt8weenQ/8FT2qYE1NXrwTG437qdmfHPXY6oo77ntFCqY6oI8Kgy9I4UK8JwNBp6qA6aOp70i300HOn5h+Ruv96H/V/L4w/Oc8WXolWnzg84acTya9h/H8xuEw5834qdAAAAABJRU5ErkJggg==\"/>\n    </div>\n</div>\n<hr>",
```
Comment (duttonw) [inline json/PDF/form.json:64]: why is the image inline?

File: src/components/partials/help-guide-lb-ingredient.js [hunk 7]
```diff
@@ -0,0 +1,1151 @@
...
+                    <span class="title">
+                        What is an ingredient list?
+                    </span>
+                    <span class="arrow"><i></i></span>
+                </label>
+            </h3>
+
+            <div class="collapsing-section" aria-hidden="true" id="id-panel-content-1">
+                <p>
+                    The ingredient list is a statement of ingredients that are used to make the food. The ingredient
```
Comment (duttonw) [inline src/components/partials/help-guide-lb-ingredient.js:35]: why is large amounts of html inside a js file, should this content be pulled from squiz as static assets or from form.io questionnaire?

**Review by duttonw — COMMENTED** [review]
Have reviewed, will wait for other to review before closing out pr (hopefully end of day monday or tuesday)

File: src/components/partials/help-guide-lb-business.js (hunk 1 above)
Comment (DanielTurner) [reply src/components/partials/help-guide-lb-business.js:1]: Partials are pulled in by the help guide component to show that part, developed separately and imported in where needed.

File: src/components/partials/help-guide-lb-exclusion.js [hunk 8]
```diff
@@ -0,0 +1,884 @@
...
+        <h3 class="acc-heading">
+            <label for="about-food">
+                <span class="title">
+                    About food names
+                </span>
+                <span class="arrow"><i></i></span>
+            </label>
+        </h3>
+
+        <div class="collapsing-section" aria-hidden="true" id="id-panel-content-1">
```
[the remark this answers is not in the document]
Comment (DanielTurner) [reply src/components/partials/help-guide-lb-exclusion.js:50]: Partials are lit-html components therefore js.

File: src/components/partials/help-guide-lb-initial.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import { html } from 'lit-html';
+
+export default (handler) => html`
+  <div class="side-padding vertical-padding">
+    <h4>
+      <i class="fa fa-info-circle"></i> Important information before you start
+    </h4>
+    <p>
+      The help guide is here to help you understand your labelling requirements.
```
[the remark this answers is not in the document]
Comment (DanielTurner) [reply src/components/partials/help-guide-lb-initial.js:9]: As above.

File: src/components/partials/help-guide-lb-main.js [hunk 10]
```diff
@@ -0,0 +1,455 @@
+import { html } from 'lit-html';
+import { modifyAccordionState } from '../../scripts/collapse-expand';
+
+export default () => html`
+      <div class="side-padding vertical-padding">
```
[the remark this answers is not in the document]
Comment (DanielTurner) [reply src/components/partials/help-guide-lb-main.js:5]: This is modern SPA frontend development.

File: src/environment.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  constructor(environment = '', base = window.location.href) {
+    let selectedEnvironment = 'production';
+    if (!environment) {
+      if (base.includes('localhost')) {
+        selectedEnvironment = 'development';
+      }
+      if (base.includes('dev')) {
+        selectedEnvironment = 'development';
+      }
+      if (base.includes('uat')) {
```
[the remark this answers is not in the document]
Comment (DanielTurner) [reply src/environment.js:15]: Sure will add a ticket for the next sprint.

File: src/environment.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    switch (selectedEnvironment) {
+      case 'development': {
+        this.url = 'https://api.forms.platforms.qld.gov.au/fesrqwsyzlbtegd/formwizard';
+        this.email = '';
+        break;
+      }
+      case 'uat': {
+        this.url = 'https://api.forms.platforms.qld.gov.au/fesrqwsyzlbtegd/formwizard';
```
[the remark this answers is not in the document]
Comment (DanielTurner) [reply src/environment.js:29]: Not sure what is meant with this.

File: src/index.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+import initialView from './components/partials/help-guide-lb-initial';
+import ingredients from './components/partials/help-guide-lb-ingredient';
+import businessView from './components/partials/help-guide-lb-business';
+import { configuration } from './config';
+import { Environment } from './environment';
+
+(() => {
+  const environment = new Environment();
+  configuration.form.location = environment.url;
+  // configuration.form.baseLocation = environment.url;
```
[the remark this answers is not in the document]
Comment (DanielTurner) [reply src/index.js:16]: It should have been deleted, not sure why it is in there.

File: src/resources/questions-ingredients/question-1/view.html (hunk 3 above)
Comment (DanielTurner) [reply src/resources/questions-ingredients/question-1/view.html:1]: It is inside the formio page, this is the component developed outside the formio environment then built and pasted in. That way we easily get around single user on formio, resolve any issues with formio backups by using a more reliable code repository.

File: json/PDF/form.json (hunk 6 above)
Comment (DanielTurner) [reply json/PDF/form.json:64]: This is a json backup of formio

File: src/components/partials/help-guide-lb-ingredient.js (hunk 7 above)
Comment (DanielTurner) [reply src/components/partials/help-guide-lb-ingredient.js:35]: That isn't how modern front end development is done.

**Comment by duttonw** [comment]
Closing out PR as its been open for 2+ working days

**Review by duttonw — APPROVED** [review]

**Pull request merged**